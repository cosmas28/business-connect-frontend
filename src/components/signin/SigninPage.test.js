import React from 'react';
import SignInPage from './SigninPage';

describe('Sign in form test', () => {
    let mountedSignInForm = null,
       props = null;
    const mockHandleSubmitForm = jest.fn(),
        mockInputChange = jest.fn();
    const component = () => {
        if (!mountedSignInForm) {
            mountedSignInForm = mount(
                <SignInPage {...props} />
            );
        }

        return mountedSignInForm;
    };

    beforeEach(() => {
        props = {
            'handleLoginInputChange': mockHandleSubmitForm,
            'handleLoginSubmitForm': mockInputChange,
            'loginPutErrorMessage': null,
            'loginSuccessMessage': null
        };
        mountedSignInForm = null;
    });

    it('renders and matches our snapshot', () => {
        expect(component()).toMatchSnapshot();
    });

    it('contains the form', () => {
        expect(component().find('input')).toHaveLength(2);
        expect(component().find('button')).toHaveLength(1);
    });

    it('should call `handleLoginInputChange` when input field is changed', () => {
        component().find('input.test').simulate('change');
        expect(props.handleLoginInputChange).toBeCalled();
    });

    it('should call `handleLoginSubmitForm` when submit button is called', () => {
        component().find('form').simulate('submit');
        expect(props.handleLoginSubmitForm).toBeCalled();
    });

    describe('when `loginSuccessMessage` is passed', () => {
        beforeEach(() => {
            props.loginSuccessMessage = 'You logged in successfully!';
        });

        it('renders a `Success Alert box`', () => {
            const alertDiv = component().find('div.alert-success');
            expect(alertDiv.length).toBe(1);
        });
    });

    describe('when `loginPutErrorMessage` is passed', () => {
        beforeEach(() => {
            props.loginPutErrorMessage = 'Invalid email or password!';
        });

        it('renders a `Error Alert box`', () => {
            const alertDiv = component().find('div.alert-danger');
            expect(alertDiv.length).toBe(1);
        });
    });

});
