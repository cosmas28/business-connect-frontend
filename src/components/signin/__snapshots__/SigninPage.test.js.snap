// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in page renders sign in page without crashing 1`] = `
<main
  className="main-content"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <SignInForm
        handleLoginInputChange={[MockFunction]}
        handleLoginSubmitForm={[MockFunction]}
      />
    </div>
  </div>
</main>
`;
